### ISSUE TITLE
Problems with vim plug

### ISSUE THREAD

**Issue by BonaBeavis** [lucc/nvimpager#8]
Should not nvimpager use its own config?

```
Error detected while processing /home/XXX/.config/nvim/init.vim:
line    5:
E117: Unknown function: plug#begin
line    7:
```

**Comment by lucc** [comment]
Which version of nvimpager are you using? What is in the config file `~/.config/nvimpager/init.vim`? Is it a symlink?

**Comment by BonaBeavis** [comment]
NVIM is  v0.3.2-dev and nvimpager is current develop. The config file does not exist.

**Comment by lucc** [comment]
Yes nvimpager should use its own config file. What do you see if you execute `bash -x ./nvimpager some-file`?

**Comment by olalonde** [comment]
Having same issue.

**Comment by lucc** [comment]
Then I have the same questions (and more) for you:
* what version of nvimpager are you running?
* what command are you running?
* how did you install it?
* do you have `~/.config/nvim` or `~/.config/nvimpager`? Is any of these a symlink?
*what is the debugging output of `bash -x nvimpager some-file`?

**Comment by cjbassi** [comment]
- `nvimpager -v`: nvimpager v0.5.1-90-g3bb4f16
- `echo foo | nvimpager`
- Installed from AUR
- Config is located at `~/.config/nvim/init.vim`
    - this is a symlink to `~/config/nvim/init.vim`
- `bash -x nvimpager .aliases`:
```
+ RUNTIME=/usr/share/nvimpager/runtime
+ export RUNTIME
+ export PPID
+ name=nvimpager
+ mode=auto
+ rc=/home/cjbassi/.config/nvimpager/init.vim
+ getopts achpv flag
+ shift 0
+ [[ ! -t 1 ]]
+ [[ ! -r /home/cjbassi/.config/nvimpager/init.vim ]]
+ rc=
+ files=()
+ [[ 1 -gt 0 ]]
+ [[ -f .aliases ]]
+ files+=("$1")
+ shift
+ [[ 0 -gt 0 ]]
+ [[ auto != pager ]]
+ [[ 1 -eq 0 ]]
+ [[ ! -t 0 ]]
+ default_args=(-R ${rc:+-u "$rc"} --cmd 'set rtp+=$RUNTIME' --cmd 'lua nvimpager = require("nvimpager")' --cmd 'lua nvimpager.stage1()' -c 'lua nvimpager.stage2()')
+ [[ auto = cat ]]
+ [[ auto = auto ]]
++ cat .aliases
++ wc -l
++ tput lines
+ [[ 152 -le 61 ]]
+ nvim -R --cmd 'set rtp+=$RUNTIME' --cmd 'lua nvimpager = require("nvimpager")' --cmd 'lua nvimpager.stage1()' -c 'lua nvimpager.stage2()' .aliases
Error detected while processing /home/cjbassi/.config/nvim/init.vim:
line    3:
E117: Unknown function: plug#begin
line    7:
E492: Not an editor command: Plug '~/drive/programming/jumplist-contained.vim'
line   13:
E492: Not an editor command: Plug 'google/vim-searchindex'
line   20:
E492: Not an editor command: Plug 'tpope/vim-endwise'
line   21:
E492: Not an editor command: Plug 'ozelentok/vim-closer'
line   26:
E492: Not an editor command: Plug 'Shougo/deoplete.nvim', { 'do': ':UpdateRemotePlugins' }
line   27:
E492: Not an editor command: Plug 'Shougo/neco-vim'
line   32:
E492: Not an editor command: Plug 'matze/vim-move'
line   33:
```
There's more but hopefully that's enough output.

I'm guessing that we need to source the vim-plug file, and more generally, any file in vim's autoload folder. For me vim-plug is location is at: `~/.local/share/nvim/site/autoload/plug.vim`

**Comment by lucc** [comment]
@cjbassi I think I found your issue. Can you try the branch `fix/issue-8`?  You will have to clone the repository but you can run nvimpager directly from there: `PAGER=~/where/ever/nvimpager`. 

I assume the errors will go away with this but nvimpager might not yet load you default nvim settings because:
You said you symlinked `~/.config/nvim/init.vim` to itself, that sounds strange. And the debug output suggests that there is no file `/home/cjbassi/.config/nvimpager/init.vim` because of these two lines
```
+ [[ ! -r /home/cjbassi/.config/nvimpager/init.vim ]]
+ rc=
```
compare https://github.com/lucc/nvimpager/blob/3bb4f16d8faca76715d4c39d8bc8eb5199476c39/nvimpager#L64-L65

If you really want to use the default nvim settings with nvimpager (I don't do it because I do not need all these plugins in nvimpager) you should probably symlink `~/.config/nvimpager` to `~/.config/nvim` and not just the init file itself.

**Comment by lucc** [comment]
@BonaBeavis the branch `fix/issue-8` might also solve the problem for you. Can you try it?

**Comment by cjbassi** [comment]
Thanks for the quick response. Few things:

Running nvimpager from the `fix/issue-8` branch doesn't error anymore, but it loads with default neovim settings I think.

Symlinking `~/.config/nvimpager` to `~/.config/nvim` causes the errors to appear again when running nvimpager from the `fix/issue-8` branch.

Also, my `init.vim` file isn't symlinked to itself, although it may look like it, because there's a dot in front of the `config` folder in the first path but not the second :D

Thanks!

**Comment by lucc** [comment]
@cjbassi Sorry I missed that dot. I have the following questions/assumptions, maybe you can answer and confirm or reject them:
1. You do not have a file `~/.config/nvimpager/init.vim`
2. Do you want nvimpager to load it's own (some special) config or your default init.vim or the neovim defaults?
3. The fact that nvimpager can not find the vim-plug functions when you symlink the whole directory is strange. I am wondering where these are installed and how you add them to the `&rtp`. If I symlink the whole folder I don't get errors (and I also use vim-plug). So:
3. Where is the vim plug file installed and how do you load it in your `~/.config/nvim/init.vim`?
4. Do you have `$IMINIT` or `$EXINIT` set in your environment? To what value?

**Comment by cjbassi** [comment]
1. Correct.
2. That's a good question. Personally, I would like to end up using my default `init.vim` but I'm not sure if nvimpager should default to that or not.
4. vim-plug is installed to `~/.local/share/nvim/site/autoload/plug.vim`. I don't do anything explicit in my `init.vim` to load it. I assume it's loaded automatically by neovim since it's located in the `autoload` folder.
5. Neither of those are set in my environment.

Do you have vim-plug installed to the same place as me?

**Comment by lucc** [comment]
2. Nvimpager will not default to the same `init.vim` file as nvim. That is a design desicion and will not change.
3. (arg markdown lists with automatic numbering)
4. nvimpager replaces all the user .../nvim/... folders with a .../nvimpager/... counterpart. So if you symlink both `~/.config/nvimpager -> ~/.config/nvim` and `~/.local/share/nvimpager/site -> ~/.local/share/nvim/site` it should work. Can you confirm that?
5. 
6. no I have installed it to `~/.local/share/nvim/plugins/vim-plug` where all the other plugins are. I add this folder to the `&rtp` in my `init.vim`.

**Comment by lucc** [comment]
I just checked the man page and it already documents these folders. Was the manpage of any help or should it be rewritten?

**Comment by cjbassi** [comment]
Cool that makes sense. So I symlinked both the `.config` and the `.local/share` folder and now it loads my settings. The man page looks good, if you want you could maybe mention the symlinking required to load the default nvim config since that might be a common thing.

So I've run into a new issue where running nvimpager on a file that is too long to print to stdout causes neovim to load the file and display it correctly, but it becomes unresponsive as soon as I press either the `j` or `k` keys and I have to force quit the application. All other movement commands work fine, but not those two. If you want I could make an new Issue for this.

**Comment by cjbassi** [comment]
So it turns out that having
```
nmap <silent> j gj
nmap <silent> k gk
```
was causing the issue, althought I'm not sure why.

**Comment by lucc** [comment]
I think this is resolved by 836962f.  If anybody has trouble when trying to use the default nvim config files please read https://github.com/lucc/nvimpager/blob/develop/nvimpager.md#CONFIGURATION before reporting.
